cmd/authd-oidc/daemon/daemon.go

```diff
@@ -125,6 +125,16 @@ func (a *App) serve(config daemonConfig) error {
 		}
 	}
 
+	brokerConfigDir := config.Paths.BrokerConf + ".d"
+	// Should these dirs be created by the server or the snap?
+	// TODO(nsklikas): are these permissions correct?
+	if err := ensureDirWithPerms(brokerConfigDir, 0700, os.Geteuid()); err != nil {
+		if err := ensureDirWithPerms(brokerConfigDir, 0755, os.Geteuid()); err != nil {
+			close(a.ready)
+			return fmt.Errorf("error initializing broker configuration directory %q: %v", brokerConfigDir, err)
+		}
+	}
+
```

Comment:
If this is true then shouldn't [this](https://github.com/ubuntu/authd-oidc-brokers/blob/main/cmd/authd-oidc/daemon/daemon.go#L122) be changed as well?

Reply:
> If this is true, then shouldn't [this](https://github.com/ubuntu/authd-oidc-brokers/blob/main/cmd/authd-oidc/daemon/daemon.go#L122) be changed as well?

No. The reason we call `ensureDirWithPerms` there twice is not that it requires different permissions to create a directory with 0700 than with 0755 (it does not), but that we want to create the data directory with 0700 by default but also support it having permissions 0755 (which is the case when it's the [`SNAP_DATA`](https://snapcraft.io/docs/data-locations#heading--system) directory, because that's created with `0755` by snap). We don't need to support that for the `broker.conf.d` directory.